modules/xfem/src/base/XFEM.C

```diff
@@ -967,6 +989,31 @@ XFEM::cutMeshWithEFA()
 
       libmesh_elem->set_node(j) = libmesh_node;
 
+      // Store solution for all nodes affected by XFEM (even existing nodes)
+      if (parent_elem->is_semilocal(_mesh->processor_id()))
```

Comment:
Wait, are you sure about this? Your processor may not own some of the nodes of a semilocal element. Sure the solution will be correct but do you attempt to write these values later?